sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionContainerTest.java

```diff
@@ -68,6 +68,17 @@ public void sessionContainer() throws Exception {
 
         sessionToken = sessionContainer.resolvePartitionLocalSessionToken(request, resolvedPKRange.getId());
         assertThat(sessionToken.getLSN()).isEqualTo(2);
+
+        String globalSessionToken = sessionContainer.resolveGlobalSessionToken(request);
```

Comment:
We have SessionTest.java where we have e2e test and way to intercept the request over the wire. Please all some crud/query operation there to validate gateway/direct mode behaving same

Reply:
The PR is intended to address primarily the ChangeFeedProcessor case where we have the partition key ID header set as a potential hot fix for users that we recommended to use Gateway mode with CFP. We will need a more broader fix that addresses all the other cases where we send the full session token when we shouldn't.